src/third_party/dartino/object_memory_mark_sweep.cc

```diff
@@ -516,31 +516,102 @@ uword CompactingVisitor::visit(HeapObject* object) {
   return size;
 }
 
-SweepingVisitor::SweepingVisitor(Program* program, OldSpace* space)
-    : HeapObjectVisitor(program), free_list_(space->free_list()), free_start_(0), used_(0) {
+// Sweep method that mostly looks at the mark bits.  For speed it doesn't touch
+// the live objects, but writes freelist structures in the gaps between them.
+uword OldSpace::sweep() {
```

Comment:
Can you validate the freelist after the sweeping? Check that a new marking doesn't lead to any bits in the free segments? Overwrite the free segments with junk to provoke errors?